src/dask_awkward/lib/core.py

```diff
@@ -68,6 +68,56 @@
 log = logging.getLogger(__name__)
 
 
+def make_dask_descriptor(func: Callable) -> Callable[[T, type[T], Array], Any]:
+    def descriptor(instance: T, owner: type[T], dask_array: Array) -> Any:
+        impl = func.__get__(instance, owner)
+        return impl(dask_array)
+
+    return descriptor
+
+
+def make_dask_method(func: Callable) -> Callable[[T, type[T], Array], Callable]:
+    def descriptor(instance: T, owner: type[T], dask_array: Array) -> Any:
+        def impl(*args, **kwargs):
+            impl = func.__get__(instance, owner)
+            return impl(dask_array, *args, **kwargs)
+
+        return impl
+
+    return descriptor
+
+
+F = TypeVar("F", bound=Callable)
+G = TypeVar("G", bound=Callable)
+
+
+class dask_property(property):
+    _dask_get: Callable | None = None
+    _dask_set: Callable | None = None
+    _dask_del: Callable | None = None
```

Comment:
I defined `__selattr__` and `__delattr__` although these wouldn't make sense in a functional API. We could come up with a `dak.setattr` and `dak.delattr` function that always returns the input array, i.e. `_dask_set` actually returns self, but I doubt we actually need `__delattr__` support.